sretoolbox/container/image.py

```diff
@@ -103,14 +107,16 @@ class Image:
 
     def __init__(self, url, tag_override=None, username=None, password=None,
                  auth_server=None, response_cache=None, auth_token=None,
-                 ssl_verify=True):
+                 ssl_verify=True, session=None, timeout=None):
         image_data = self._parse_image_url(url)
         self.scheme = image_data['scheme']
         self.registry = image_data['registry']
         self.repository = image_data['repository']
         self.image = image_data['image']
         self.response_cache = response_cache
         self.ssl_verify = ssl_verify
+        self.session = session
```

Comment:
Why would we create a session outside of this class? I would make more sense to have the session created inside...

Reply:
Because if this class create the session, then it has to close it in `__exit__`, `Image` has to be used in contextmanager like

```python
with Image() as image:
    ...
```

Reply:
Sounds reasonable. If you want a session, use contextmanager. Otherwise, no session. At least we manage our own session.

Reply:
it becomes hard once I see `__getitem__` is returning new `Image`, better to keep `__init__` side effect free to let caller manage lifecycle of session.

Reply:
that's exactly why we never had a session here :)
alright, your call. tks for the consideration.